contracts/protocols/mainnet/compound_iii/helpers.sol

```diff
@@ -0,0 +1,440 @@
+        TotalsBasic memory _totalsBasic = _comet.totalsBasic();
+        uint8 _numAssets = _comet.numAssets();
+        address baseTokenPriceFeed = _comet.baseTokenPriceFeed();
+
+        int256 amount_ = int256(
+            (currentValue(
+                _userBasic.principal,
+                _totalsBasic.baseSupplyIndex,
+                _totalsBasic.baseBorrowIndex,
+                _comet.baseIndexScale()
+            ) * int256(_comet.getPrice(baseTokenPriceFeed))) / int256(1e8)
+        );
+
+        for (uint8 i = 0; i < _numAssets; i++) {
+            if (isAssetIn(_userBasic.assetsIn, i)) {
+                AssetInfo memory asset = _comet.getAssetInfo(i);
+                UserCollateral memory coll = _comet.userCollateral(account, asset.asset);
+                uint256 newAmount = (uint256(coll.balance) * _comet.getPrice(asset.priceFeed)) / 1e8;
+                amount_ += int256((newAmount * asset.borrowCollateralFactor) / 1e18);
+            }
+        }
+
+        return amount_;
+    }
+
+    function getAccountFlags(address account, IComet _comet) internal view returns (AccountFlags memory flags) {
+        flags.isLiquidatable = _comet.isLiquidatable(account);
+        flags.isBorrowCollateralized = _comet.isBorrowCollateralized(account);
+    }
+
+    function getCollateralData(
+        address account,
+        IComet _comet,
+        uint8[] memory offsets
+    ) internal returns (UserCollateralData[] memory _collaterals, address[] memory collateralAssets) {
+        UserBasic memory _userBasic = _comet.userBasic(account);
+        uint16 _assetsIn = _userBasic.assetsIn;
+        uint8 numAssets = uint8(offsets.length);
+        Token memory _token;
+        uint8 _length = 0;
+
+        for (uint8 i = 0; i < numAssets; i++) {
+            if (isAssetIn(_assetsIn, offsets[i])) {
+                _length++;
+            }
+        }
+        _collaterals = new UserCollateralData[](_length);
+        collateralAssets = new address[](_length);
+        uint8 j = 0;
+
+        for (uint8 i = 0; i < numAssets; i++) {
+            if (isAssetIn(_assetsIn, offsets[i])) {
+                AssetInfo memory asset = _comet.getAssetInfo(offsets[i]);
+                _token.token = asset.asset;
+                _token.symbol = TokenInterface(asset.asset).symbol();
+                _token.decimals = TokenInterface(asset.asset).decimals();
+                _token.scale = asset.scale;
+                _token.offset = asset.offset;
+
+                uint256 suppliedAmt = uint256(_comet.userCollateral(account, asset.asset).balance);
+                _collaterals[j].token = _token;
+                collateralAssets[j] = _token.token;
+                _collaterals[j].suppliedBalanceInAsset = suppliedAmt;
+                _collaterals[j].suppliedBalanceInBase = getCollateralBalanceInBase(
+                    suppliedAmt,
+                    _comet,
+                    asset.priceFeed
+                );
+                j++;
```

Comment:
@Richa-iitr If the user hasn't entered into the market, then you won't add that collateral?

Reply:
If the user doesn't have collateral then it will not show in user's collateral data but the market collaterals/assets will be shown in market configuration as assets.

Reply:
Okay!